packages/webcomponents/src/components/payments-list/payments-list-core.tsx

```diff
@@ -0,0 +1,122 @@
+import { Component, Host, h, Prop, State, Watch, Event, EventEmitter } from '@stencil/core';
+import { PagingInfo, Payment, pagingDefaults } from '../../api';
+import { MapPaymentStatusToBadge, formatCurrency, formatDate, formatTime } from '../../utils/utils';
+
+@Component({
+  tag: 'payments-list-core',
+  styleUrl: 'payments-list.scss',
+})
+
+export class PaymentsListCore {
+  @Prop() getPayments: Function;
+  @State() payments: Payment[] = [];
+  @State() loading: boolean = true;
+  @State() errorMessage: string;
+  @State() paging: PagingInfo = pagingDefaults;
+  @State() params: any;
+  @Event({
+    eventName: 'payment-row-clicked',
+
+    bubbles: true,
+  }) rowClicked: EventEmitter<Payment>;
+
+  componentWillLoad() {
+    if (typeof this.getPayments === 'function') {
```

Comment:
I'm not totally sure I understand this line 24 - is this so that test spec files won't try to call `fetchData`? 🤔

Reply:
This is just to prevent `fetchData` from being called in case no `getPayments` is passed (which is only in test files) 

This can be simplified to just `if (this.getPayments)`. I'll change that and also remove the redundant validation on line 50. Good catch!